pkg/server/plugin/datastore/sql/sql.go

```diff
@@ -1079,19 +1093,43 @@ func deleteRegistrationEntry(tx *gorm.DB,
 		return nil, err
 	}
 
-	if err := tx.Model(&entry).Association("FederatesWith").Clear().Error; err != nil {
+	err = deleteRegistrationEntrySupport(tx, entry)
+	if err != nil {
 		return nil, err
 	}
 
-	if err := tx.Delete(&entry).Error; err != nil {
-		return nil, sqlError.Wrap(err)
-	}
-
 	return &datastore.DeleteRegistrationEntryResponse{
 		Entry: respEntry,
 	}, nil
 }
 
+func deleteRegistrationEntrySupport(tx *gorm.DB, entry RegisteredEntry) error {
+	if err := tx.Model(&entry).Association("FederatesWith").Clear().Error; err != nil {
+		return err
+	}
+
+	if err := tx.Delete(&entry).Error; err != nil {
+		return sqlError.Wrap(err)
+	}
+
+	return nil
+}
+
+func pruneRegistrationEntries(tx *gorm.DB, req *datastore.PruneRegistrationEntriesRequest) (*datastore.PruneRegistrationEntriesResponse, error) {
+	var registrationEntries []RegisteredEntry
+	if err := tx.Where("expiry != 0").Where("expiry <= ?", req.ExpiresBefore).Find(&registrationEntries).Error; err != nil {
```

Comment:
the field name ExpiresBefore suggests this should be a `<`?

Reply:
I was wondering about that; `pruneJoinTokens` has this same logic, so I think both should be changed to `<` not `<=`. Would that be alright?

Reply:
pushed change assuming "yes" to above :)